flink-runtime/src/main/java/org/apache/flink/runtime/io/network/buffer/Buffer.java

```diff
@@ -343,7 +343,7 @@ public static DataType getDataType(AbstractEvent event, boolean hasPriority) {
                 return EVENT_BUFFER;
             }
             CheckpointBarrier barrier = (CheckpointBarrier) event;
-            if (barrier.getCheckpointOptions().needsAlignment()) {
+            if (barrier.getCheckpointOptions().isExactlyOnceMode()) {
```

Comment:
?

Reply:
A non-prioritized UC currently doesn't revoke credits. However, actually the barrier now needs to be aligned, so I guess that change is bad.

Probably, it would be cleaner to encode the temporary non-UC UC behavior into the `CheckpointOptions`. Maybe by transforming `isUnalignedCheckpoint` into an enum `ALIGNED, UNALIGNED, TEMP_ALIGNED`.

Reply:
Implemented it as discussed in the second half.